usersession/agent/rest_api.go

```diff
@@ -323,6 +323,8 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 		})
 	}
 
+	// XXX: instead of closing the open notifications show a notification
+	// that just says that the snap is now ready to use?
```

Comment:
I presume that it should do both things: close the old notification, and show a new notification that says that the snap is ready to be used...

Reply:
I'm trying to do that in another branch, but it seems that this is never called...

Reply:
This is a suggestion to do that:
```
diff --git a/usersession/agent/rest_api.go b/usersession/agent/rest_api.go
index 157393f6b2..a76b6170eb 100644
--- a/usersession/agent/rest_api.go
+++ b/usersession/agent/rest_api.go
@@ -300,6 +300,34 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
        return SyncResponse(nil)
 }
 
+func sendDesktopStandardNotification(notificationMgr notification.NotificationManager, appName string, 
instanceName string, summary string, body string, hints []notification.Hint, appDesktopEntry string) err
or {
+       var icon string
+
+       // The notification is provided by snapd session agent.
+       hints = append(hints, notification.WithDesktopEntry("io.snapcraft.SessionAgent"))
+       // But if we have a desktop file of the busy application, use that apps's icon.
+       if appDesktopEntry != "" {
+               parser := goconfigparser.New()
+               desktopFilePath := filepath.Join(dirs.SnapDesktopFilesDir, appDesktopEntry+".desktop")
+               if err := parser.ReadFile(desktopFilePath); err == nil {
+                       icon, _ = parser.Get("Desktop Entry", "Icon")
+               }
+       }
+
+       msg := &notification.Message{
+               AppName: appName,
+               Title:   summary,
+               Icon:    icon,
+               Body:    body,
+               Hints:   hints,
+       }
+       if err := notificationMgr.SendNotification(notification.ID(instanceName), msg); err != nil {
+               logger.Noticef("unable to send a standard notification, %s, for app %s", summary, appName)
+               return err
+       }
+       return nil
+}
+
 func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
        if ok, resp := validateJSONRequest(r); !ok {
                return resp
@@ -326,13 +354,22 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
        // XXX: instead of closing the open notifications show a notification
        // that just says that the snap is now ready to use?
        if err := c.s.notificationMgr.CloseNotification(notification.ID(finishRefresh.InstanceName)); err != nil {
+               logger.Noticef("cannot send close notification message: %v", err)
+       }
+
+       summary := fmt.Sprintf(i18n.G("Refreshed “%s” snap"), finishRefresh.InstanceName)
+       body := i18n.G("Ready to launch.")
+
+
+       var hints []notification.Hint
+       if err := sendDesktopStandardNotification(c.s.notificationMgr, finishRefresh.InstanceName, finishRefresh.InstanceName, summary, body, hints, ""); err != nil {
                return SyncResponse(&resp{
                        Type:   ResponseTypeError,
                        Status: 500,
                        Result: &errorResult{
-                               Message: fmt.Sprintf("cannot send close notification message: %v", err),
+                               Message: fmt.Sprintf("cannot send refresh done notification message: %v", err),
                        },
                })
        }
+
```

Reply:
but the editor doesn't allow me to add it as a change suggestion.

Reply:
There was one bug in the user agent rest api handler that got fixed in this PR now, essentially `finishRefreshNotificationCmd` was missing from `restApi` so that was never reached :(

Reply:
I mark this resolved because there is a PR that tracks this suggestion